[docs-only]  added the details to the 'Restore Trash-Bins Items' doc section' (#9228)
owncloud/ocis: services/storage-users/README.md

Example cases for expired uploads:

> *   In the final step the upload blob is moved from the upload area to the final blobstore (e.g. S3).

Reviewer: ```suggestion
*   In the final step, the upload blob is moved from the upload area to the final blobstore (e.g. S3).
```

The behaviour of the `purge-expired` command can be configured by using the following environment variables.

> *   `STORAGE_USERS_PURGE_TRASH_BIN_USER_ID`

Reviewer: ```suggestion
*   `STORAGE_USERS_PURGE_TRASH_BIN_USER_ID`\
```
The double blanks at the end were on purpose as they tell the renderer to put the following text into the next line. Double blanks at the ned can be replaced alternatively with a `\`.

Used to obtain space trash-bin information and takes the system admin user as the default which is the `OCIS_ADMIN_USER_ID` but can be set individually. It should be noted, that the `OCIS_ADMIN_USER_ID` is only assigned automatically when using the single binary deployment and must be manually assigned in all other deployments. The command only considers spaces to which the assigned user has access and delete permission.

> *   `STORAGE_USERS_PURGE_TRASH_BIN_PERSONAL_DELETE_BEFORE`

Reviewer: ```suggestion
*   `STORAGE_USERS_PURGE_TRASH_BIN_PERSONAL_DELETE_BEFORE`\
```

Has a default value of `720h` which equals `30 days`. This means, the command will delete all files older than `30 days`. The value is human-readable, for valid values see the duration type described in the [Environment Variable Types](https://doc.owncloud.com/ocis/5.0/deployment/services/envvar-types-description.html). A value of `0` is equivalent to disable and prevents the deletion of `personal space` trash-bin files.

> *   `STORAGE_USERS_PURGE_TRASH_BIN_PROJECT_DELETE_BEFORE`

Reviewer: ```suggestion
*   `STORAGE_USERS_PURGE_TRASH_BIN_PROJECT_DELETE_BEFORE`\
```

Has a default value of `720h` which equals `30 days`. This means, the command will delete all files older than `30 days`. The value is human-readable, for valid values see the duration type described in the [Environment Variable Types](https://doc.owncloud.com/ocis/5.0/deployment/services/envvar-types-description.html). A value of `0` is equivalent to disable and prevents the deletion of `project space` trash-bin files.

#### List and Restore Trash-Bins Items

> Restoring is possible only at the original location. The personal or project spase ID `spaceID` is required for the items restoring.

Reviewer 2: ```suggestion
Restoring is possible only to the original location. The personal or project space ID `spaceID` is required for the items to be restored.
```

> To authenticate the cli tool use `OCIS_SERVICE_ACCOUNT_SECRET=<acc-secret>` and `OCIS_SERVICE_ACCOUNT_ID=<acc-id>`. The `storage-users` cli tool uses the default address to establish the connection to the `gateway` service. If the connection is failed check your custom `gateway`

Reviewer 2: ```suggestion
To authenticate the cli tool use `OCIS_SERVICE_ACCOUNT_SECRET=<acc-secret>` and `OCIS_SERVICE_ACCOUNT_ID=<acc-id>`. The `storage-users` cli tool uses the default address to establish the connection to the `gateway` service. If the connection fails check your custom `gateway`
```

service `GATEWAY_GRPC_ADDR` configuration and set the same address to `storage-users` variable `OCIS_GATEWAY_GRPC_ADDR` or `STORAGE_USERS_GATEWAY_GRPC_ADDR`.

*   Print a list of all trash-bin items of a space
    ```bash

>     ocis storage-users trash-bin list [command options] ['spaceID' required]

Reviewer: For the example, we should add the envvars named above to make the usage and need clear.

Author: Added the export example. I think  good enough https://github.com/owncloud/ocis/pull/9228/commits/aae2947660d09f9420ed2b4bf1efed9bb50cce62

    ```

> The restore option defines the behavior for an item to be restored, when the item name already exists in the target space. Supported options are: `skip`, `replace` and `keep-both`. The default value is `skip`

Reviewer: ```suggestion
The restore option defines the behavior for an item to be restored, when the item name already exists in the target space. Supported options are: `skip`, `replace` and `keep-both`. The default value is `skip`.

```

> When the cli tool restores the item with the `replace` option, the existing item will be moved to a trash-bin. When the cli tool restores the item with the `keep-both` option and the designation item already exists, the name of the restored item will be changed by adding the numeric suffix in parentheses. The variable `STORAGE_USERS_CLI_MAX_ATTEMPTS_RENAME_FILE` defines a maximum number of attempts to rename an item.

Reviewer 2: ```suggestion
When the cli tool restores the item with the `replace` option, the existing item will be moved to a trash-bin. When the cli tool restores the item with the `keep-both` option and the designated item already exists, the name of the restored item will be changed by adding a numeric suffix in parentheses. The variable `STORAGE_USERS_CLI_MAX_ATTEMPTS_RENAME_FILE` defines a maximum number of attempts to rename an item.
```

* Restore all trash-bin items for a space
    ```bash
    ocis storage-users trash-bin restore-all [command options] ['spaceID' required]
    ```

* Restore a trash-bin item by ID
    ```bash

>     ocis storage-users trash-bin restore [command options] ['spaceID' required] ['itemID' required]

Reviewer: Add needed envvars as suggested above.

To authenticate the cli tool use `OCIS_SERVICE_ACCOUNT_SECRET=<acc-secret>` and `OCIS_SERVICE_ACCOUNT_ID=<acc-id>`. The `storage-users` cli tool uses the default address to establish the connection to the `gateway` service. If the connection fails, check your custom `gateway`
service `GATEWAY_GRPC_ADDR` configuration and set the same address to `storage-users` variable `OCIS_GATEWAY_GRPC_ADDR` or `STORAGE_USERS_GATEWAY_GRPC_ADDR`.

*   Export the gateway address if your configuration differs from the default
    ```bash

>     export STORAGE_USERS_GATEWAY_GRPC_ADDR=gateway:9142

Reviewer: Q: is that really `gateway` or better `127.0.0.1` ?
Because `gateway` comes from the HelmChart and I guess that they have used a variable...
